Src/AutoFakeItEasy/FakeItEasyBuilder.cs

```diff
@@ -73,16 +73,24 @@ public object Create(object request, ISpecimenContext context)
 #pragma warning restore 618
             }
 
-            var fake = this.builder.Create(request, context) as FakeItEasy.Configuration.IHideObjectMembers;
+            var fake = this.builder.Create(request, context);
             if (fake == null)
             {
 #pragma warning disable 618
                 return new NoSpecimen(request);
 #pragma warning restore 618
             }
 
+            var createdType = fake.GetType();
+            if (!createdType.IsFake())
+            {
+#pragma warning disable 618
+                return new NoSpecimen(request);
+#pragma warning restore 618
+            }
+
             var fakeType = type.GetFakedType();
-            if (fake.GetType().GetFakedType() != fakeType)
+            if (createdType.GetFakedType() != fakeType)

```

Comment:
Could lines 76-93 be refactored so that we avoid calling `GetFakedType` twice? :smiley: